ocaml-eglot-type-enclosing.el

```diff
@@ -0,0 +1,171 @@
+            (1- (length ocaml-eglot-type-enclosing-types)))
+        (setq ocaml-eglot-type-enclosing-offset 0)
+      (setq ocaml-eglot-type-enclosing-offset
+            (1+ ocaml-eglot-type-enclosing-offset)))
+    (ocaml-eglot-type-enclosing--with-fixed-offset)))
+
+(defun ocaml-eglot-type-enclosing-shrink ()
+  "Display the type enclosing of a smaller enclosing if possible."
+  (interactive)
+  (when ocaml-eglot-type-enclosing-types
+    (if (<= ocaml-eglot-type-enclosing-offset 0)
+        (setq ocaml-eglot-type-enclosing-offset
+              (1- (length ocaml-eglot-type-enclosing-types)))
+      (setq ocaml-eglot-type-enclosing-offset
+            (1- ocaml-eglot-type-enclosing-offset)))
+    (ocaml-eglot-type-enclosing--with-fixed-offset)))
+
+(defun ocaml-eglot-type-enclosing--type-buffer (type-expr)
+  "Create buffer with content TYPE-EXPR of the enclosing type buffer."
+  (let ((curr-dir default-directory)
+        (current-major-mode major-mode))
+    (with-current-buffer (get-buffer-create ocaml-eglot-type-buffer-name)
+      (funcall current-major-mode)
+      (read-only-mode 0)
+      (erase-buffer)
+      (insert type-expr)
+      (goto-char (point-min))
+      (read-only-mode 1)
+      (setq default-directory curr-dir))))
+
+(defun ocaml-eglot-type-enclosing--display (type-expr &optional current-enclosing)
+  "Display the type-enclosing for TYPE-EXPR in a dedicated buffer.
+If CURRENT-ENCLOSING is set, the range of the enclosing will be highlighted."
+  (ocaml-eglot-type-enclosing--type-buffer type-expr)
+  (if (ocaml-eglot-util--text-less-than type-expr 8)
+      (message "%s" (with-current-buffer ocaml-eglot-type-buffer-name
+                      (font-lock-fontify-region (point-min) (point-max))
+                      (buffer-string)))
+    (display-buffer ocaml-eglot-type-buffer-name))
+  (when (and current-enclosing (> (length ocaml-eglot-type-enclosing-types) 0))
+    (let ((current-enclosing (aref ocaml-eglot-type-enclosing-types
+                                   ocaml-eglot-type-enclosing-offset)))
+      (ocaml-eglot-util--highlight-range current-enclosing
+                                         'ocaml-eglot-highlight-region-face))))
+
+(defun ocaml-eglot-type-enclosing--reset ()
+  "Reset local variables defined by the enclosing query."
+  (setq ocaml-eglot-type-enclosing-current-type nil)
+  (setq ocaml-eglot-type-enclosing-verbosity nil)
+  (setq ocaml-eglot-type-enclosing-types nil)
+  (setq ocaml-eglot-type-enclosing-offset 0))
+
+(defun ocaml-eglot-type-enclosing--call ()
+  "Print the type of the expression under point."
+  (ocaml-eglot-type-enclosing--reset)
+  (let* ((verbosity nil)
+         (index 0)
+         (at (ocaml-eglot-util--current-position-or-range))
+         (result (ocaml-eglot-req--type-enclosings at index verbosity))
+         (type (cl-getf result :type))
+         (enclosings (cl-getf result :enclosings)))
+    (setq ocaml-eglot-type-enclosing-verbosity nil)
```

Comment:
Why not using the `verbosity` value ?

Also, since reset was called, and verbosity is always `nil` here, that looks like a no-op isn't it ?

Same remark for `index`.